docs/spec/lightclient/detection/detection.md

```diff
@@ -363,8 +560,92 @@ The detector shall maintain the following invariants:
 and the following transition invariant
    - *FullNodes' \union Secondaries' \union FaultyNodes' = FullNodes \union Secondaries \union FaultyNodes*
 
+
+> The following invariant is very useful for reasoning, and underlies
+> many intuition when we 
+
+#### **[LCD-INV-TRUSTED-AGREED.1]:**
+
+It is always the case the light client has downloaded a lightblock for height
+*lightStore.LatestTrusted().Height* 
+from each of the current primary and the secondary, that all reported
+the identical lightblock for that height.
+
+> In the above, I guess "the identical" might be replaced with "a
+> matching" to cover commits that might be different.
+
+> The above requires us that before we pick a new secondary, we have to
+> query the secondary for the header of height 
+> *lightStore.LatestTrusted().Height*.
```

Comment:
Note that this doesn't ensure that trusted store of the secondary is consistent with the primary, i.e., only the latest trusted light block must be consistent. Could this potentially be problematic if secondary (with invalid store except latest trusted light block) is later promoted to a primary?

Reply:
This is subtle. First, this current supervisor is simplified as it considers only going forward (`VerifyToTarget` only goes forward). In this case, it is not a problem. 

If we want the supervisor to be more general it need to use [Main](https://github.com/informalsystems/tendermint-rs/blob/master/docs/spec/lightclient/verification/verification.md#lcv-func-ibcmain1) instead of `VerifyToTarget`. Then we would need to query the (new) primary for an older trusted block as reference. This older trusted block then also needs to be passed into Forkdetection (which also assumes going forward). This is less critical as we discuss to anyway only pass in one block in the future. @OStevan has prepared a Rust code snippet and will create a PR on data structures and what information to pass around.